spiceaidocs/docs/components/data-connectors/clickhouse.md

```diff
@@ -16,20 +14,50 @@ datasets:
+| `clickhouse_connection_string` | The connection string to use to connect to the Clickhouse server. This can be used instead of providing individual connection parameters.                                                                                                                                                                               |
+| `clickhouse_host`              | The hostname of the Clickhouse server.                                                                                                                                                                                                                                                                                  |
+| `clickhouse_tcp_port`          | The port of the Clickhouse server.                                                                                                                                                                                                                                                                                      |
+| `clickhouse_db`                | The name of the database to connect to.                                                                                                                                                                                                                                                                                 |
+| `clickhouse_user`              | The username to connect with.                                                                                                                                                                                                                                                                                           |
+| `clickhouse_pass`              | The password to connect with.                                                                                                                                                                                                                                                                                           |
+| `clickhouse_secure`            | Optional. Specifies the SSL/TLS behavior for the connection, supported values:<br /> <ul><li>`true`: (default) This mode requires an SSL connection. If a secure connection cannot be established, server will not connect.</li><li>`false`: This mode will not attempt to use an SSL connection, even if the server supports it.</li></ul> |
+| `connection_timeout`           | Optional. Specifies the connection timeout in milliseconds.                                                                                                                                                                                                                                                             |
```

Comment:
This should support specifying the timeout in the same format as our other timeout parameters, in a human readable format like `10s` or `1m`.

Reply:
I don't think we need to do this now, but we should at least create an issue to track it, since its inconsistent with our other connectors.

Reply:
Issue created to track the fix: https://github.com/spiceai/spiceai/issues/3710